core/shared/src/main/scala/io/circe/Decoder.scala

```diff
@@ -771,10 +775,7 @@ final object Decoder extends TupleDecoders with ProductDecoders with LowPriority
   implicit final val monadInstances: SemigroupK[Decoder] with MonadError[Decoder, DecodingFailure] =
     new SemigroupK[Decoder] with MonadError[Decoder, DecodingFailure] {
       final def combineK[A](x: Decoder[A], y: Decoder[A]): Decoder[A] = x.or(y)
-      final def pure[A](a: A): Decoder[A] = new Decoder[A] {
-        final def apply(c: HCursor): Result[A] = Xor.right(a)
-        override def decodeAccumulating(c: HCursor): AccumulatingDecoder.Result[A] = Validated.valid(a)
-      }
+      final def pure[A](a: A): Decoder[A] = const(a)
```

Comment:
you could add `pureEval` right, and only evaluate on `apply`? That laziness might be useful for some folks.